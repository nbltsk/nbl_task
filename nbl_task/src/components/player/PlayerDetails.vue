<template>
  <div class="player-details">
    <ul class="player-details__list">
      <li
        v-for="(value, name) in details"
        :key="`value-${name}`"
        class="player-details__one"
      >
        <span>{{ name | capitalize }}: {{ value }}</span>
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import filters from '../../mixins/filters';
import { Player } from "../../types/player.model";
export default Vue.extend({
  name: 'PlayerDetails',
  mixins: [filters],
  props: {
    details: {
      type: Object as PropType<Player>,
      required: true,
    },
  },
});
</script>
<style lang="scss">
.player-details {
  display: inline-block;

  &__list {
    display: inline-flex;
    flex-direction: column;
  }

  &__one {
    font-size: 20px;
    line-height: 28px;
    letter-spacing: 0.2px;
  }
}
</style>