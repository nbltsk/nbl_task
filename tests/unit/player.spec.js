import PlayerDetails from '../../src/components/player/PlayerDetails';
import { mount } from '@vue/test-utils';

test('Snapshot test for PlayerDetails', () => {
  const wrapper = mount(PlayerDetails, {
    propsData: {
      details: {
        id: 123,
        first_name: 'Michael',
        height_feet: 3,
        height_inches: 5,
        last_name: 'Jordan',
        position: 'Shooting guard',
        team: 'Chicago Bulls',
        weight_pounds: 50
      }
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('Check if PlayerDetails is rendering properly', () => {
  const wrapper = mount(PlayerDetails, {
    propsData: {
      details: {
        id: 123,
        first_name: 'Michael',
        height_feet: 3,
        height_inches: 5,
        last_name: 'Jordan',
        position: 'Shooting guard',
        team: 'Chicago Bulls',
        weight_pounds: 50
      }
    }
  });
  const playerDetails = wrapper.find('.player-details__one');
  expect(playerDetails.exists()).toBe(true);
  expect(wrapper.html()).toContain('Michael');
  expect(wrapper.html()).toContain('<li class="player-details__one"><span>Team: Chicago Bulls</span></li>');
});