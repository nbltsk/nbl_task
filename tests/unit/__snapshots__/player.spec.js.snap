// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for PlayerDetails 1`] = `
<div class="player-details">
  <ul class="player-details__list">
    <li class="player-details__one"><span>Id: 123</span></li>
    <li class="player-details__one"><span>First name: Michael</span></li>
    <li class="player-details__one"><span>Height feet: 3</span></li>
    <li class="player-details__one"><span>Height inches: 5</span></li>
    <li class="player-details__one"><span>Last name: Jordan</span></li>
    <li class="player-details__one"><span>Position: Shooting guard</span></li>
    <li class="player-details__one"><span>Team: Chicago Bulls</span></li>
    <li class="player-details__one"><span>Weight pounds: 50</span></li>
  </ul>
</div>
`;
